client/webserver/site/src/js/markets.js

```diff
@@ -225,6 +230,73 @@ export default class MarketsPage extends BasePage {
     Doc.hide(page.loaderMsg)
   }
 
+  /* setRegistrationStatusViewContent sets the text content and class for the
+   * registration status view
+   */
+  setRegistrationStatusView (titleContent, messageContent, titleClass) {
+    const page = this.page
+    page.regStatusTitle.textContent = titleContent
+    page.regStatusMessage.textContent = messageContent
+    page.regStatusTitle.classList.remove('completed', 'error', 'waiting')
+    page.regStatusTitle.classList.add(titleClass)
+  }
+
+  /*
+   * updateRegistrationStatusView udpates the view based on the current
+   * registration status
+   */
+  updateRegistrationStatusView (dexUrl, feePaid, confirmationsRequired, confirmations) {
+    const page = this.page
+
+    page.confReq.textContent = confirmationsRequired
+    page.regStatusDex.textContent = dexUrl
+
+    if (feePaid) {
+      this.setRegistrationStatusView('Registration fee payment successful!', '', 'completed')
+      return
+    }
+
+    const waitingMessage = [confirmations, confirmationsRequired]
+      .every(v => typeof v === 'number') ? `${confirmations} / ${confirmationsRequired}` : ''
```

Comment:
`confirmationsRequired` will always be a number now, I think.

Reply:
ok, I removed this type check.